bokeh/charts/_charts.py

```diff
@@ -144,17 +147,44 @@ def start_plot(self, xgrid, ygrid):
         if ygrid:
             self.make_grid(1, yaxis.ticker)
 
-        # Add tools
+        # Add tools if supposed to
         if self.tools:
+            # need to add tool to all underlying plots
             for plot in self._plots:
+                removed_tools = []
+                # only add tools if the underlying plot hasn't been customized
+                # by some user injection
                 if not plot.tools:
-                    if not self.categorical:
-                        pan = PanTool()
-                        wheelzoom = WheelZoomTool()
-                        reset = ResetTool()
-                        plot.add_tools(pan, wheelzoom, reset)
-                    previewsave = PreviewSaveTool()
-                    plot.add_tools(previewsave)
+                    tools = []
+                    # if tools have been specified as string let's reuse
+                    # plotting machinery to create the related tools
+                    if isinstance(self.tools, string_types):
+                        for tool in re.split(r"\s*,\s*", self.tools.strip()):
+                            # re.split will return empty strings; ignore them.
+                            if tool == "":
+                                continue
```

Comment:
Yes in general there should be deliberation, but in this particular case it seems pretty straightforward: I think there should be a function that takes a `Plot` and a `tools` argument. It converts the `tools` argument to a list of `Tool` objects and assigns that to `Plot.tools` (or possible returns the list for the caller to assign). This should be fine unless there is some functionality of behavior that is different between the APIs that I am overlooking.